tests/tape/test_tape.py

```diff
@@ -1299,3 +1299,115 @@ def test_casting(self):
+            with tape.stop_recording():
+                pass
+
+    def test_nested_tape_not_recording(self):
+        """Test that an error is raised if the tape is no
+        longer recording"""
+        with QuantumTape() as tape1:
+            qml.RX(0.1, wires=0)
+
+            with QuantumTape() as tape2:
+                qml.RX(0.1, wires=0)
+
+                with pytest.raises(qml.queuing.QueuingError, match="Cannot stop recording"):
+                    with tape1.stop_recording():
+                        pass
+
+    def test_recording_stopped(self):
+        """Test that recording is stopped within a tape context"""
+
+        with QuantumTape() as tape:
+            op0 = qml.RX(0, wires=0)
+            assert tape.active_context() is tape
+
+            with tape.stop_recording():
+                op1 = qml.RY(1.0, wires=1)
+                assert tape.active_context() is None
+
+            op2 = qml.RZ(2, wires=1)
+            assert tape.active_context() is tape
+
+        assert len(tape.operations) == 2
+        assert tape.operations[0] == op0
+        assert tape.operations[1] == op2
+
+    def test_nested_recording_stopped(self):
+        """Test that recording is stopped within a nested tape context"""
+
+        with QuantumTape() as tape1:
+            op0 = qml.RX(0, wires=0)
+            assert tape1.active_context() is tape1
+
+            with QuantumTape() as tape2:
+                assert tape1.active_context() is tape2
+                op1 = qml.RY(1.0, wires=1)
+
+                with tape2.stop_recording():
+                    assert tape1.active_context() is None
+                    op2 = qml.RZ(0.6, wires=2)
+                    op3 = qml.CNOT(wires=[0, 2])
+
+                op4 = qml.Hadamard(wires=0)
+
+            op5 = qml.RZ(2, wires=1)
+            assert tape1.active_context() is tape1
+
+        assert len(tape1.operations) == 3
+        assert tape1.operations[0] == op0
+        assert tape1.operations[1] == tape2
+        assert tape1.operations[2] == op5
+
+        assert len(tape2.operations) == 2
+        assert tape2.operations[0] == op1
+        assert tape2.operations[1] == op4
+
+    def test_creating_scratch_tape(self):
+        """Test that a tape created inside the 'scratch'
+        space is properly created and accessible"""
+        with QuantumTape() as tape:
+            op0 = qml.RX(0, wires=0)
+            assert tape.active_context() is tape
+
+            with tape.stop_recording(), QuantumTape() as temp_tape:
+                assert tape.active_context() is temp_tape
+                op1 = qml.RY(1.0, wires=1)
+
+            op2 = qml.RZ(2, wires=1)
+            assert tape.active_context() is tape
+
+        assert len(tape.operations) == 2
+        assert tape.operations[0] == op0
+        assert tape.operations[1] == op2
+
+        assert len(temp_tape.operations) == 1
+        assert temp_tape.operations[0] == op1
+
```

Comment:
Can you add a test where you use `stop_recording` twice? so like
```python
with tape1.stop_recording():
   with tape2.stop_recording():
      ...
```

Reply:
I did originally (I had tests for this, and more complicated internal logic where I was removing and inserting from an index of the `deque` queue), but then realized it doesn't really make sense -- the way tapes currently work, only the most deeply nested tape is ever recording operations. Does that make sense?